build_runner/lib/src/package_graph/apply_builders.dart

```diff
@@ -65,13 +94,37 @@ class BuilderApplication {
 /// dependency on some other package by choosing the appropriate
 /// [BuilderApplication].
 List<BuildAction> createBuildActions(PackageGraph packageGraph,
-        Iterable<BuilderApplication> builderApplications) =>
-    stronglyConnectedComponents<String, PackageNode>(
-            [packageGraph.root], (node) => node.name, (node) => node.dependencies)
-        .expand((cycle) => builderApplications.expand((b) => cycle
-            .where(b.filter)
-            .map((p) => new BuildAction(b.builder, p.name,
-                inputs: b.inputs,
-                excludes: b.excludes,
-                isOptional: b.isOptional))))
-        .toList();
+    Iterable<BuilderApplication> builderApplications, List<String> args) {
```

Comment:
These args aren't always going to be the same for all packages/builderApplications... how do we handle that?

Reply:
We'll deal with that once we add support for modules and manual config.

The important thing here was just that we can deal with it without a breaking signature change.